internal/bloblang/query/methods_structured_test.go

```diff
@@ -149,3 +150,382 @@ func TestMethodImmutability(t *testing.T) {
+			input:    []any{1, "x", "x", 2},
+			param:    "x",
+			expected: []any{[]any{1}, []any{}, []any{2}},
+		},
+		{
+			name:     "array split start and end delimiters",
+			input:    []any{"x", 1, 2, "x"},
+			param:    "x",
+			expected: []any{[]any(nil), []any{1, 2}, []any{}},
+		},
+
+		// Byte array splitting
+		{
+			name:     "bytes split with comma",
+			input:    []byte("foo,bar,baz"),
+			param:    ",",
+			expected: []any{[]byte("foo"), []byte("bar"), []byte("baz")},
+		},
+		{
+			name:     "bytes split empty input",
+			input:    []byte(""),
+			param:    ",",
+			expected: []any{[]byte("")},
+		},
+		{
+			name:     "string split start delimiter",
+			input:    ",hello,world",
+			param:    ",",
+			expected: []any{"", "hello", "world"},
+		},
+		{
+			name:     "string split end delimiter",
+			input:    "hello,world,",
+			param:    ",",
+			expected: []any{"hello", "world", ""},
+		},
+		{
+			name:     "array split all elements match",
+			input:    []any{"x", "x", "x"},
+			param:    "x",
+			expected: []any{[]any(nil), []any{}, []any{}, []any{}},
+		},
+
+		// Error cases
+		{
+			name:   "invalid input type",
+			input:  123,
+			param:  ",",
+			errMsg: "expected string, array or bytes value",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fn, err := InitMethodHelper("split", NewLiteralFunction("", tt.input), tt.param)
+			require.NoError(t, err)
+
+			result, err := fn.Exec(ctx)
+
+			if tt.errMsg != "" {
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), tt.errMsg)
+				return
+			}
+
+			require.NoError(t, err)
+			assert.Equal(t, tt.expected, result)
+		})
+	}
+}
+
+func TestSplitByMethod(t *testing.T) {
+	// Helper functions for creating predicates
+	literalFn := func(val any) Function {
+		return NewLiteralFunction("", val)
+	}
+
+	arithmetic := func(left, right Function, op ArithmeticOperator) Function {
+		t.Helper()
+		fn, err := NewArithmeticExpression(
+			[]Function{left, right},
+			[]ArithmeticOperator{op},
+		)
+		require.NoError(t, err)
+		return fn
+	}
+
+	methods := func(fn Function, methods ...struct {
+		name string
+		args []any
+	}) Function {
+		t.Helper()
+		for _, m := range methods {
+			var err error
+			fn, err = InitMethodHelper(m.name, fn, m.args...)
+			require.NoError(t, err)
+		}
+		return fn
+	}
+
+	method := func(name string, args ...any) struct {
+		name string
+		args []any
+	} {
+		return struct {
+			name string
+			args []any
+		}{name: name, args: args}
+	}
+
+	ctx := FunctionContext{
+		Index: 0,
+		MsgBatch: message.Batch{
+			message.NewPart([]byte("test")),
+		},
+	}
+
+	tests := []struct {
+		name      string
+		input     any
+		predicate Function
+		expected  any
+		errMsg    string
+	}{
+		{
+			name:  "string split by space characters",
+			input: "hello world test",
+			predicate: arithmetic(
+				NewFieldFunction(""),
+				literalFn(" "),
+				ArithmeticEq,
+			),
+			expected: []any{"hello", "world", "test"},
+		},
+		{
```

Comment:
```suggestion
		},
		{
			name:  "string with emoji split by space characters",
			input: "❤️ 🧡 💛 💚 💙 💜",
			predicate: arithmetic(
				NewFieldFunction(""),
				literalFn("💛"),
				ArithmeticEq,
			),
			expected: []any{"❤️ 🧡 ", " 💚 💙 💜"},
		},
		{
			name:  "chinese split by space characters",
			input: "你好 世界",
			predicate: arithmetic(
				NewFieldFunction(""),
				literalFn(" "),
				ArithmeticEq,
			),
			expected: []any{"你好", "世界"},
		},
		{
```

The "string with emoji split by emoji" will catch issues with UTF-8 & Runes but it passes as-is 😎